Projects/Server/Items/Container.Enumerable.cs

```diff
@@ -0,0 +1,226 @@
+                    }
+
+                    if (item is Container { m_Items.Count: > 0 } c)
+                    {
+                        containersForGetItem.Enqueue(c);
+                    }
+                }
+            }
+            while (containersForGetItem.TryDequeue(out current));
+
+            containersForGetItem.Clear();
+
+            return @default;
+        }
+
+        public bool Any() => FirstOrDefault() != null;
+
+        public PooledRefQueue<TResult> ToPooledRefQueue()
+        {
+            PooledRefQueue<TResult> toRet = new();
+
+            foreach (TResult item in this)
+            {
+                toRet.Enqueue(item);
+            }
+
+            return toRet;
+        }
+
+        public List<TResult> ToList()
+        {
+            List<TResult> toRet = new();
+
+            foreach (TResult item in this)
+            {
+                toRet.Add(item);
+            }
+
+            return toRet;
+        }
+
+        public TestAllItemsEnumerator<TFilter, TResult> GetEnumerator() => new(container, in filter);
+    }
+
+    public ref struct TestAllItemsEnumerator<TFilter, TResult>
+        where TFilter : IFilter<Item, TResult>
+    {
+        private readonly PooledRefQueue<Container> containers;
+        private readonly ref TFilter filter;
+        private Span<Item> items;
+        private int index;
+
+        public TResult Current { get; private set; }
+
+        public TestAllItemsEnumerator(Container container, in TFilter filter)
+        {
+            this.filter = filter;
+
+            containers = PooledRefQueue<Container>.Create();
+
+            if (container.m_Items is not null)
+            {
+                items = CollectionsMarshal.AsSpan(container.m_Items);
+            }
+
+            Current = default!;
+        }
+
+        public bool MoveNext()
+        {
+            return SetNextItem() || SetNextContainer() && SetNextItem();
+        }
+
+        private bool SetNextContainer()
+        {
+            if (!containers.TryDequeue(out Container? c))
+            {
+                return false;
+            }
+
+            items = CollectionsMarshal.AsSpan(c.m_Items);
+            index = 0;
+            return true;
+        }
+
+        private bool SetNextItem()
+        {
+            while (index < items.Length)
+            {
+                Item item = items[index++];
+                if (item is Container { m_Items.Count: > 0 } c)
+                {
+                    containers.Enqueue(c);
+                }
+
+                TResult? convertedItem = filter.Invoke(item);
+                if (convertedItem is not null)
+                {
+                    Current = convertedItem;
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        public readonly void Dispose()
+        {
+            containers.Dispose();
+        }
+    }
+
+    public ref struct TestItemsEnumerator<TFilter, TResult>
```

Comment:
Looks like the difference between TestAllItems and TestItems is recursiveness. Should we use a flag instead of two implementations?

Reply:
Another miss...the name of this iterator is wrong 😄 
The implementation for recursive and non recursive is quite different: non-recursive don't need queues as buffer and have less code statements, so I thought to separate entirely both and expose the non-recursive on a different set.

If you want I still can introduce the flag as the standard RunUO implementation and use a single path for both cases